openghg/retrieve/_search_helpers.py

```diff
@@ -0,0 +1,138 @@
+"""
+Helper functions for processing search input into form accepted by `Metastore.search`
+"""
+
+import itertools
+from typing import Any, cast, Callable, Optional, TypeVar
+from openghg.util import extract_float
+from openghg.types import Comparable
+
+
+T = TypeVar("T", bound=Comparable)  # types with <=
+
+
+def _in_interval(x: T, start: Optional[T], stop: Optional[T]) -> bool:
+    """Return True if start <= x <= stop; if either start or stop is None, omit
+    the corresponding bound.
+
+    Args:
+        x: value to test for inclusion in the interval [start, stop]
+        start: start of the interval
+        stop: end of the interval
+
+    Returns:
+        True if start <= x <= stop, False otherwise
+    """
+    if start is None and stop is None:
+        return False
```

Comment:
This case should return `True` to be consistent with other uses of `None` in this this function